Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -366,15 +366,15 @@ extension HTTPClient {
     /// `EventLoopFuture<Response>` of the execution or cancellation of the execution.
     public final class Task<Response> {
         /// `EventLoop` used to execute and process this request.
-        public let eventLoop: EventLoop
+        public internal(set) var currentEventLoop: EventLoop
```

Comment:
Now that this is a variable, we must not allow mutation from the public scope.
I thought it might be the right time to set it to `currentEventLoop` as discussed in the recent comments in #82 now that this property is a variable, but I can also make the name change in another commit if preferred.
cc @weissi @artemredkin

Reply:
This looks good but we need some synchronisation here because it is mutable might be accessed from different threads now. I'd suggest to depend on `NIOConcurrencyHelpers` and store it in

    private let _currentEventLoop: AtomicBox<EventLoop>
    public var currentEventLoop: EventLoop {
        return self._currentEventLoop.load()
    }
```

Reply:
Why do we need it to be `var`? I somehow missed it

Reply:
It's because of `setChannel(...)`: when it updates the channel, it must also update the current event loop at the same time.